astropy/coordinates/representation.py

```diff
@@ -483,7 +554,141 @@ def represent_as(self, other_class):
             return self
         else:
             # The default is to convert via cartesian coordinates
-            return other_class.from_cartesian(self.to_cartesian())
+            if isiterable(other_class):
+                if isinstance(other_class, six.string_types):
+                    raise ValueError('Input to a representation\'s represent_as'
+                                     ' must be a class, not a string. For '
+                                     'strings, use frame objects')
+                if (self.differentials is None or
+                        len(other_class) != 1+len(self.differentials)):
```

Comment:
For a dict, I think there would be the choice to give just two classes if there is only one differential, or a class for the representation and a dict for the differentials if there is more than one.